theme/scripts/components/Component.ts

```diff
@@ -26,6 +26,23 @@ export default class Component<D> {
 		this.init()
 	}
 
+	protected getChild<C extends HTMLElement>(
+		selector: string,
+		F: Constructor<C>,
+		parent: HTMLElement = this.el
+	): C {
+		const child = parent.querySelector(selector)
+
+		if (!(child instanceof F)) {
+			throw new ComponentInitializationError()
```

Comment:
Method can be called out of `init`, so the error naming (`...InitializationError`) is not accurate 🤡